Modules/Core/Controllers/UsersController.php

```
@@ -186,13 +189,42 @@ public function changePassword(string $user_id)
+        if (!$user) {
+            return $this->redirectWithError('users.index', TranslationHelper::trans('user_not_found'));
+        }
+
+        // Business rule: Cannot delete users with related invoices, quotes, or sessions
+        if (!$this->userService->canDelete($userId)) {
+            $blockers = $this->userService->getDeletionBlockers($userId);
+            $message = TranslationHelper::trans('user_deletion_not_allowed', [
+                'invoices' => $blockers['invoices'],
+                'quotes'   => $blockers['quotes'],
+                'sessions' => $blockers['sessions'],
+            ]);
```

PHPMD (2.15.0)

Avoid using static access to class '\Modules\Core\Support\TranslationHelper' in method 'delete'. (undefined)

(StaticAccess)